src/Services/AuthService.ts

```diff
@@ -0,0 +1,121 @@
+import { authEndpoint, clientId, redirectUri } from "../Constants/Constants"
+import { getAuthenticationToken } from "../Endpoints/Authorization"
+import { AuthToken } from "../Models/Authentication"
+import { GetTokenRequest } from "../Models/Requests/AuthenticationRequests"
+
+// class that handles all authentication
+export default class PKCEAuthentication {
```

Comment:
Another nitpick, this should probably either be named `AuthService` or the filename should change to `PKCEAuthService`. `PKCEAuthentication` is a verb not a noun so it's hard for me to think of it as an object